src/ui/public/directives/render_complete.js

```diff
@@ -0,0 +1,40 @@
+import { uiModules } from 'ui/modules';
+
+const attributeName = 'data-render-complete';
+
+uiModules
+  .get('kibana')
+  .directive('renderComplete', () => ({
+    controller($scope, $element) {
+      const el = $element[0];
+
+      const start = () => {
+        $element.attr(attributeName, false);
+        return true;
+      };
+
+      const complete = () => {
+        $element.attr(attributeName, true);
+        return true;
+      };
+
+      const teardown = () => {
```

Comment:
'$scope.$on('$destroy', treardown)` ?

Reply:
When `setup` is called it's doing exactly this: ` $scope.$on('$destroy', teardown);` am I missing something?

Reply:
nope, im blind :) sorry